CMakeLists.txt

```diff
@@ -70,6 +70,7 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "WASI")
         $<$<COMPILE_LANGUAGE:C,CXX>:_WASI_EMULATED_MMAN>)
     add_link_options("-Lwasi-emulated-signal")
     add_link_options("-Lwasi-emulated-mman")
+elseif(CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
```

Comment:
(nit) Could you add a comment here explaining it's not needed on `FreeBSD`? So it looks clear this is intentional.